### DISCUSSION TITLE
No demo recording

### DISCUSSION THREAD

**Discussion by Bunker56** [PhlexPlexico/G5API#236] [Help]
I apologize for the trouble, but a question has arisen - a demo is not being written on get 5 v.0.11. get5.cfg has the get5_demo_upload_url parameter. Does it have anything to do with recording a demo? The get5_upload_demos 1 parameter is set.  There are no errors in the logs.  All plugins are the latest versions. Backups are written normally.

**Comment by PhlexPlexico** [comment]
No, you can ignore the new demo implementation of get5, as my solution is different. If your demos are not uploading, make sure you are waiting for the demo upload to complete. Larger demos can take time if they are being uploaded from a non-local network. In your API log you should see a hit to `match/<matchid>/demo/upload` during the start, and it should give a `200` once completed.

**Reply by Bunker56** [reply]
Now I get this:
L 12/07/2022 - 02:23:40: [get5] Calling Get5_OnDemoFinished()
L 12/07/2022 - 02:23:40: [G5WS] About to enter UploadDemo. SO YES WE ARE. Our match ID is 4
L 12/07/2022 - 02:23:40: [G5WS] Trying to create request to url match/4/map/0/demo
L 12/07/2022 - 02:23:40: [G5WS] Added http header value key=01BU9IJUQ4NEZTS95AD0OVEY to request
L 12/07/2022 - 02:23:40: [G5WS] Adding the API Key to the body is deprecated. Will be replaced once G5API 1.7.0 is released.
L 12/07/2022 - 02:23:40: [G5WS] Added param key=01BU9IJUQ4NEZTS95AD0OVEY to request
L 12/07/2022 - 02:23:40: [G5WS] Our api url: http://mmr.bunker56.ru/api/
L 12/07/2022 - 02:23:40: [G5WS] Our demo string: 2022-12-07_02-17-35_4_map1_de_mirage.dem
L 12/07/2022 - 02:23:40: [G5WS] Added param demoFile=2022-12-07_02-17-35_4_map1_de_mirage.dem to request
src/common/httpclient.cpp (796) : pHost && *pHost
src/common/httpclient.cpp (796) : pHost && *pHost
L 12/07/2022 - 02:23:40: [G5WS] Trying to create request to url match/4/map/0/demo/upload
assert_20221207022340_15.dmp[104485]: Uploading dump (out-of-process)
/tmp/dumps/assert_20221207022340_15.dmp
assert_20221207022340_15.dmp[104485]: Finished uploading minidump (out-of-process): success = no
assert_20221207022340_15.dmp[104485]: error: libcurl.so: cannot open shared object file: No such file or directory
assert_20221207022340_15.dmp[104485]: file ''/tmp/dumps/assert_20221207022340_15.dmp'', upload no: ''libcurl.so: cannot open shared object file: No such file or directory''
L 12/07/2022 - 02:23:40: [G5WS] Added http header value key=01BU9IJUQ4NEZTS95AD0OVEY to request
L 12/07/2022 - 02:23:40: [G5WS] Uploading demo to server...
src/common/httpclient.cpp (796) : pHost && *pHost
src/common/httpclient.cpp (796) : pHost && *pHost

Sourcemod 1.11,6923  Roll back to 1.10?

**Reply by PhlexPlexico** [reply]
Sure, you can always attempt to roll back, but I already ran into this issue once. What os is your game server on, and does it have the 32 bit version of libcurl installed?

**Reply by Bunker56** [reply]
> Sure, you can always attempt to roll back, but I already ran into this issue once. What os is your game server on, and does it have the 32 bit version of libcurl installed?

Debian 11, 32 bit version of libcurl installed, Or try get5 0.10?

**Reply by PhlexPlexico** [reply]
I would try 0.10 and see if the demo does upload. I am wondering if this is a new version issue, or existed prior to and no one has noticed it.

Thanks!

**Reply by Bunker56** [reply]
SM 1.11.6924. last G5WS, get5 0.10.5 - working
SM 1.11.6924, last G5WS, last get5 0.11 - problems with uploading demo. May be get5_demo_upload_url or get5_demo_path?
SM 1.10.6545,last G5WS. last get5 0.11 - crash server. Ooops!..

**Reply by Bunker56** [reply]
I would like to use get5 0.11, there are interesting features in it...

**Reply by PhlexPlexico** [reply]
There's inherent issues with demo recording. Make sure your server has enough RAM allocated to it as well. Anything with `tv_delay` higher than 0 will automatically allocate 2GB of RAM. You'll have to bring up the issue with the get5 repo if there is crashing issues with get5. Try without demo recording `tv_enable 0` and see if the server crashes. This is more of a server configuration problem, and not to do with the API or plugin.

**Reply by Bunker56** [reply]
I will still use version 0.10, it is guaranteed to work, and solve the problem in the process...
